students/alex_whitty/lesson4/trigrams.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+
+with open('double_take.txt', 'r') as f:
+
+    for line in f:
+        print(line, end='')
+
+
+
+
+trigrams = {("When", "the"): ['Travelers'],
+            ("the", "Travelers"): ["from"],
+            ("Travelers", "from"): ["Outer"],
+            ("from", "Outer"): ["space"],
+            ("Outer", "Space"): ["dug"],
+            ("Space", "dug"): ["into"],
+            ("dug", "into"): ["the"],
+            ("into", "the"): ["pile"],
+            }
+
+
+
+def build_trigrams(words):
+
+    trigrams = {}
+
+    for i in range(len(words) - 2):
+
+
+        pair = tuple(words[i:i + 2])
+        follower = words[i + 2]
+
+        if pair not in trigrams:
+
+            trigrams[pair] = [follower]
+        else:
+            trigrams[pair].append(follower)
+
+
+    return trigrams
+
+# build up the dict here!
+
+def build_dict(words):
+
+    import random
```

Comment:
Don't do the import from inside of a function. This should be at the top of the file, along with other _import_ statements.